test/AdoNetCore.AseClient.Tests/Integration/StreamingTests.cs

```diff
@@ -0,0 +1,117 @@
+using System;
+using System.Collections.Generic;
+using System.Data;
+using AdoNetCore.AseClient.Internal;
+using Dapper;
+using NUnit.Framework;
+
+namespace AdoNetCore.AseClient.Tests.Integration
+{
+    [TestFixture]
+    [Category("extra")]
+    public class StreamingTests
+    {
+        private static string CleanUp =
+@"IF EXISTS(SELECT 1 FROM sysobjects WHERE name = 'stream-result-sets')
+BEGIN
+    DROP PROCEDURE [dbo].[stream-result-sets]
+END";
+        private static string TestProc =
+@"CREATE PROCEDURE [dbo].[stream-result-sets]
+(
+    @sleepTime CHAR(8),
+    @numberOfLoops INT
+)
+AS
+BEGIN
+    DECLARE @index INT
+    SET @index = 0
+
+    WHILE (@index < @numberOfLoops)
+    BEGIN
+        SELECT @index AS [Index], GETUTCDATE() AS Now
+
+        SET @index = @index + 1
+        WAITFOR DELAY @sleepTime
+    END
+END
+";
+        private AseConnection GetConnection()
+        {
+            Logger.Enable();
+            return new AseConnection(ConnectionStrings.Pooled);
+        }
+
+        [SetUp]
+        public void Setup()
+        {
+            using (var connection = GetConnection())
+            {
+                connection.Execute(CleanUp);
+                connection.Execute(TestProc);
+            }
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            using (var connection = GetConnection())
+            {
+                connection.Execute(CleanUp);
+            }
+        }
+
+        // TODO - a failing test that shows the result sets cannot be streamed.
+        // TODO - use this to implement an IEnumerable socket, an IEnumerable token reader, and make data reader work accordingly.
+        [Test]
+        public void DelayedResultSetTests()
```

Comment:
I tried to run this test locally but it kept on failing because the expected/actual values differed by about 4 seconds. Is that an issue?

Reply:
That's the feature I'm trying to add. Still can't make it work yet though.